File: services/go/models/comment.go
package models

import "time"

type Comment struct {
	ID        int       `json:"id"`
	PostID    int       `json:"post_id"`
	UserID    int       `json:"user_id"`
	Content   string    `json:"content"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

func NewComment(postID, userID int, content string) *Comment {
	return &Comment{
		PostID:    postID,
		UserID:    userID,
		Content:   content,
		CreatedAt: time.Now(),
		UpdatedAt: time.Now(),
	}
}

func (c *Comment) Update(content string) {
	c.Content = content
	c.UpdatedAt = time.Now()
}

func (c *Comment) ToMap() map[string]interface{} {
	return map[string]interface{}{
		"id":         c.ID,
		"post_id":    c.PostID,
		"user_id":    c.UserID,
		"content":    c.Content,
		"created_at": c.CreatedAt,
		"updated_at": c.UpdatedAt,
	}
}

File: services/ruby/models/comment.rb
module Models
  class Comment
    attr_accessor :id, :post_id, :user_id, :content, :created_at, :updated_at

    def initialize(attributes = {})
      @id = attributes[:id]
      @post_id = attributes[:post_id]
      @user_id = attributes[:user_id]
      @content = attributes[:content]
      @created_at = attributes[:created_at] || Time.now
      @updated_at = attributes[:updated_at] || Time.now
    end

    def to_h
      {
        id: @id,
        post_id: @post_id,
        user_id: @user_id,
        content: @content,
        created_at: @created_at,
        updated_at: @updated_at
      }
    end

    def to_json(*args)
      to_h.to_json(*args)
    end
  end
end
